crates/orderbook/openapi.yml

```diff
@@ -540,32 +530,113 @@ components:
+        sellAmount:
+          description: see `OrderParameters::sellAmount`
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+        buyAmount:
+          description: see `OrderParameters::buyAmount`
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+        validTo:
+          description: see `OrderParameters::validTo`
+          type: integer
+        feeAmount:
+          description: see `OrderParameters::feeAmount`
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+        kind:
+          description: see `OrderParameters::kind`
+          allOf:
+            - $ref: "#/components/schemas/OrderKind"
+        partiallyFillable:
+          description: see `OrderParameters::partiallyFillable`
+          type: boolean
+        sellTokenBalance:
+          description: see `OrderParameters::sellTokenBalance`
+          allOf:
+            - $ref: "#/components/schemas/SellTokenSource"
+          default: "erc20"
+        buyTokenBalance:
+          description: see `OrderParameters::buyTokenBalance`
+          allOf:
+            - $ref: "#/components/schemas/BuyTokenDestination"
+          default: "erc20"
+        signingScheme:
+          $ref: "#/components/schemas/SigningScheme"
+        signature:
+          $ref: "#/components/schemas/Signature"
+        from:
+          description: |
+            If set, the backend enforces that this address matches what is decoded as the signer of
+            the signature. This helps catch errors with invalid signature encodings as the backend
+            might otherwise silently work with an unexpected address that for example does not have
+            any balance.
+          allOf:
+            - $ref: "#/components/schemas/Address"
+          nullable: true
+        quoteId:
+          description: |
+            Orders can optionally include a quote ID. This way the order can be linked to a quote
+            and enable providing more metadata when analyzing order slippage.
+          type: integer
+          nullable: true
+        appData:
+          description: |
+            This field comes in two forms for backward compatibility. The first form (app data hash) will eventually stop being accepted.
+          anyOf:
+            - $ref: "#/components/schemas/AppDataHash"
+            - title: Full App Data
               description: |
-                Orders can optionally include a quote ID. This way the order can be linked to a quote
-                and enable providing more metadata when analyzing order slippage.
-              type: integer
-              nullable: true
-          required:
-            - signingScheme
-            - signature
+                Short:
+
+                If you do not care about app data, set this field to `\"{}\"` and make sure that the order you signed for this request had its app data field set to `0xb48d38f93eaa084033fc5970bf96e559c33c4cdc07d889ab00b4d63f9590739d`.
```

Comment:
I don't think we should encourage people not not use this by saying they can add `{}`. Ideally, integrations should create one that would include their appCode at least. 

Also, If they don't care, why not leaving it blank if you don't care about it? `0x0` and undefined appData?

Reply:
> But from a backend API point of view, it has no meaning

That's the thing, the appData is not meant to have meaning for the backend (exclusively)
It's meant for adding information that doesn't fit into what backend or the SC expects.

I think at least the `appCode` is already valuable information for Analytics to discern source of orders?
cc @bh2smith to confirm whether this is used ATM.

> The frontend happens to follow some app data specification but we're not forcing everyone to follow it.

We on the contrary would like everyone to follow it, not for frontend's sake, but for everyone interacting with the protocol.
This way you know the format and what type of data will be stored, how to read it, etc

> Either way should be easy to use for an api user that doesn't care about app data

I also agree with Anxo that having a null condition rather than `{}` is preferred

Reply:
I've rewritten the "revamp app data" issue to talk about frontend vs backend app data more.

Reply:
> I don't know how we use the frontend's app data at the moment. But from a backend API point of view, it has no meaning

@vkgnosis I have to say, I mentioned this before, and I never like when we make this differentiation Backend/Frontend, and you call this front-end appData. We really made an effort to make this part very explicit.

Let me explain clarify my point.

When we defined this, even though it was a Frontend initiative (we were the first creating orders), we brought @bh2smith @alfetopito @alfetopito and @nlordell into the discussion. We discussed over a document until we got an agreement. This is a soft contract (not SC enforced), but is highly beneficial of encouraging people to use a common patter/specification for the meta-data

We agreed on a format that can be future proof:
- Allows to add new types of metadata
- It's versioned: both the document, and the metadatas
- Its definition is formalized in https://json-schema.org/
- Its in GIT https://github.com/cowprotocol/app-data so you can see the schemas, and its evolution

We provided tutorials: https://docs.cow.fi/front-end/creating-app-ids/create-the-order-meta-data-file

We created Notion pages: 
- https://www.notion.so/cownation/appData-Orders-Metadata-020d5f4e18bd45248a26ec5b16dba6df?pvs=4
- https://www.notion.so/cownation/App-data-Old-way-to-create-c133f1d27acc40838aef5cb8ad54a66d?pvs=4

The Data team did an integration and is now consumed in Dune.

Some external parts use this format. 

It is used for the affiliate program (now deprecated), not only in CoW Swap, but external parties

It is used the UTM codes https://www.notion.so/cownation/Create-UTM-codes-for-on-chain-analytics-cd1ec4ce33964da48a374c8d17b00913?pvs=4

I belive is not a Frontend standard, its something we put a lot of effort into having some agreement on this format, is something we should advertise even though it was not meaningful for the backend until pre-interactions was not implemented.

This can be changed, and it's good to see how we improve it (the hashing change is a good idea), but I want to make sure we agree this is not Frontend and whatever we do needs to be:
- Carefully planned
- Carefully Agreed by all teams
- Shoudn't break any of the apps, documentation, etc (all this should be part of the plan)

Reply:
Thanks for clarifying that. I didn't know about the history.

Reply:
No worries @vkgnosis I understand, happy I could provide more context.